File: implementations/hashset/csharp/Hashset.cs
using System;


namespace HashSet
{
    public class Node
    {
        public int val;
        public Node next;

        public Node(int _val)
        {
            this.val = _val;
        }
    }
    
    /// <summary>
    /// Leetcode 705. Design Hashset
    /// https://leetcode.com/problems/design-hashset/
    /// </summary>
    public class MyHashSet
    {
        private const int ARRAY_SIZE = 100;
        private Node[] nodes = new Node[ARRAY_SIZE];

        public void Add(int value)
        {
            var index = value % ARRAY_SIZE;
            var head = nodes[index];

            if (head == null)
            {
                nodes[index] = new Node(value);
                return;
            }

            if (head.val == value)
                return;
            
            while (head.next != null)
            {
                if (head.next.val == value)
                    return;
                
                head = head.next;
            }

            head.next = new Node(value);
        }

        public bool Contains(int value)
        {
            var index = value % ARRAY_SIZE;

            var head = nodes[index];

            if (head == null)
                return false;

            while (head != null)
            {
                if (head.val == value)
                    return true;
                
                head = head.next;
            }
            
            return false;
        }

        public void Remove(int value)
        {
            var index = value % ARRAY_SIZE;
            var head = nodes[index];

            if (head == null)
                return;

            if (head.val == value)
            {
                nodes[index] = nodes[index].next;
                return;
            }

            var current = head;
            while (current.next != null)
            {
                if (current.next.val == value)
                {
                    current.next = current.next.next;
                    return;
                }

                current = current.next;
            }
        }
    }
}

File: implementations/HashSet/go/hashset.go
package hashset

type listNode struct {
	val  int
	next *listNode
}

type MyHashSet struct {
	buckets []*listNode
}

func Constructor() MyHashSet {
	bckts := make([]*listNode, 100)
	return MyHashSet{buckets: bckts}
}

func (this *MyHashSet) Add(key int) {
	var bucketIndex = key % len(this.buckets)
	var bucket = this.buckets[bucketIndex]

	if bucket == nil {
		bucket = &listNode{val: key}
		this.buckets[bucketIndex] = bucket
		return
	}

	if bucket.val == key {
		return
	}

	for bucket.next != nil {
		//already contains
		if bucket.val == key {
			return
		}
		bucket = bucket.next
	}
	bucket.next = &listNode{val: key}
}

func (this *MyHashSet) Remove(key int) {
	var bucketIndex = key % len(this.buckets)
	var bucket = this.buckets[bucketIndex]

	//means there's no such value
	if bucket == nil {
		return
	}

	//if it's head
	if bucket.val == key {
		bucket = bucket.next
		this.buckets[bucketIndex] = bucket
		return
	}

	for bucket.next != nil {
		if bucket.next.val == key {
			bucket.next = bucket.next.next
			return
		}

		bucket = bucket.next
	}

}

func (this *MyHashSet) Contains(key int) bool {
	var bucketIndex = key % len(this.buckets)
	var bucket = this.buckets[bucketIndex]

	if bucket == nil {
		return false
	}

	for bucket != nil {
		if bucket.val == key {
			return true
		}
	return false
}
